tests/drivers/build_all/rtc/i2c_devices.overlay

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024 TOKITA Hiroshi
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	test {
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		test_gpio: gpio@deadbeef {
+			compatible = "vnd,gpio";
+			gpio-controller;
+			reg = <0xdeadbeef 0x1000>;
+			#gpio-cells = <0x2>;
+			status = "okay";
+		};
+
+		test_i2c: i2c@11112222 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "vnd,i2c";
+			reg = <0x11112222 0x1000>;
+			status = "okay";
+			clock-frequency = <100000>;
+
+			test_am1805: am1805@0 {
+				compatible = "ambiq,am1805";
+				status = "okay";
+				reg = <0x0>;
+				am1805-gpios = <&test_gpio 0 0>;
+			};
+
+			test_pcf8523: pcf8523@1 {
+				compatible = "nxp,pcf8523";
+				status = "okay";
+				reg = <0x1>;
+				alarms-count = <1>;
+				battery-switch-over = "standard";
+				int1-gpios = <&test_gpio 13 0>;
```

Comment:
nit: this will probably help avoid people scratching their head in the future as to why "13" :) (or maybe I am missing something, which is very much a possibility!)

```suggestion
				int1-gpios = <&test_gpio 0 0>;
```